### PR TITLE
Clothing armor value tag on examine from /tg/

### PR DESCRIPTION
**Pull request by BrumalLake**
## About The Pull Request
Ports https://github.com/tgstation/tgstation/pull/49571
Inspecting gives a little link you can click to see an expanded list of armor values for a piece of clothing
planning on doing weapon damage as well soon

## Why It's Good For The Game
Less tabbing out and diving into the code for 3 minutes when you see a new cool piece of clothing and want to see exactly how good its protection is

## Changelog
:cl:
add: Ability to see a tag with the armor values of a piece of clothing when you inspect it
/:cl:

### CODE REVIEW

**Comment by francinum** [comment]
Ah yes, time to charge into battle with my burn prot 5 armor.

I don't entirely agree that disseminating armor values is a good idea, but I also acknowledge that it's either this or codediving.

**Comment by utilbird** [comment]
Yeah it's probably better to have this as in game info so everyone's on the same playing field for dmg/def value ""metagaming""

**Comment by ItsTisme** [comment]
I think its nice, lets you actuely look between 2 clothing items and see which one to wear if your for example sec. or just in general

**Comment by PowerfulBacon** [comment]
How does it look in action?

**Comment by BrumalLake** [comment]
> 
> 
> How does it look in action?

theres a little link in chat you click and it prints out a list of about how much armor it has

**Comment by francinum** [comment]
I'm still very much *NOT* a fan of the idea of this, but if it's this or raw values, I'll tolerate this.

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -159,6 +152,89 @@
...
+
+	if(LAZYLEN(durability_list))
+		durability_list.Cut()
+	if(armor.fire)
+		durability_list += list("FIRE" = armor.fire)
+	if(armor.acid)
+		durability_list += list("ACID" = armor.acid)
+
+	if(LAZYLEN(armor_list) || LAZYLEN(durability_list))
+		. += "<span class='notice'>It has a <a href='?src=[REF(src)];list_armor=1'>tag</a> listing its protection classes.</span>"
```
Comment (PowerfulBacon) [inline code/modules/clothing/clothing.dm]: This proabably doesn't need to be done every single time on examine

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -159,6 +152,89 @@
...
+			. = "VI"
+		if (7)
+			. = "VII"
+		if (8)
+			. = "VIII"
+		if (9)
+			. = "IX"
+		if (10 to INFINITY)
+			. = "X"
+	return .
```
Comment (PowerfulBacon) [inline code/modules/clothing/clothing.dm]: return . is not needed as . is the default return value

File: code/modules/clothing/clothing.dm [hunk 3]
```diff
@@ -159,6 +152,89 @@
...
+  * Rounds armor_value to nearest 10, divides it by 10
+  * and then expresses it in roman numerals up to 10
+  * Arguments:
+  * * armor_value - Number we're converting
+  */
+/obj/item/clothing/proc/armor_to_protection_class(armor_value)
+	armor_value = round(armor_value,10) / 10
+	switch (armor_value)
+		if (1)
+			. = "I"
```
Comment (PowerfulBacon) [inline code/modules/clothing/clothing.dm]: What if armor_value is 4 (or negative), in which case null will be returned and the display will be "BOMB "

File: code/modules/clothing/clothing.dm [hunk 4]
```diff
@@ -159,6 +152,89 @@
...
+
+	if(LAZYLEN(armor_list) || LAZYLEN(durability_list))
+		. += "<span class='notice'>It has a <a href='?src=[REF(src)];list_armor=1'>tag</a> listing its protection classes.</span>"
+
+/obj/item/clothing/Topic(href, href_list)
+	. = ..()
+
+	if(href_list["list_armor"])
+		var/list/readout = list("<span class='notice'><u><b>PROTECTION CLASSES (I-X)</u></b>")
+		if(LAZYLEN(armor_list))
```
Comment (PowerfulBacon) [inline code/modules/clothing/clothing.dm]: This shouldn't be lazylen if the list isn't lazy.

File: code/modules/clothing/clothing.dm [hunk 5]
```diff
@@ -159,6 +152,89 @@
...
+			readout += "\n<b>ARMOR</b>"
+			for(var/dam_type in armor_list)
+				var/armor_amount = armor_list[dam_type]
+				readout += "\n[dam_type] [armor_to_protection_class(armor_amount)]" //e.g. BOMB IV
+		if(LAZYLEN(durability_list))
```
Comment (PowerfulBacon) [inline code/modules/clothing/clothing.dm]: This shouldn't be lazylen if the list isn't lazy.

File: code/modules/clothing/clothing.dm [hunk 6]
```diff
@@ -159,6 +152,89 @@
 		how_cool_are_your_threads += "</span>"
 		. += how_cool_are_your_threads.Join()
 
+	if(LAZYLEN(armor_list))
```
Comment (PowerfulBacon) [inline code/modules/clothing/clothing.dm]: This shouldn't be lazylen if the list isn't lazy.

File: code/modules/clothing/clothing.dm [hunk 7]
```diff
@@ -159,6 +152,89 @@
...
+	if(armor.magic)
+		armor_list += list("MAGIC" = armor.magic)
+	if(armor.melee)
+		armor_list += list("MELEE" = armor.melee)
+	if(armor.rad)
+		armor_list += list("RADIATION" = armor.rad)
+	if(armor.stamina)
+		armor_list += list("STAMINA" = armor.stamina)
+
+	if(LAZYLEN(durability_list))
```
Comment (PowerfulBacon) [inline code/modules/clothing/clothing.dm]: This shouldn't be lazylen if the list isn't lazy.

File: code/modules/clothing/clothing.dm [hunk 8]
```diff
@@ -159,6 +152,89 @@
...
+		durability_list.Cut()
+	if(armor.fire)
+		durability_list += list("FIRE" = armor.fire)
+	if(armor.acid)
+		durability_list += list("ACID" = armor.acid)
+
+	if(LAZYLEN(armor_list) || LAZYLEN(durability_list))
```
Comment (PowerfulBacon) [inline code/modules/clothing/clothing.dm]: This shouldn't be lazylen if the list isn't lazy.

File: code/modules/clothing/clothing.dm (hunk 1 above)
Comment (BrumalLake) [reply code/modules/clothing/clothing.dm]: how else would it gather that the lists have enough length to tell you that the object has a tag on it

**Comment by BrumalLake** [comment]
done with gags so i'll tweak this in a bit

now only generates when you click it and this is what it says when a value is less than 5
image
only thing is that it's a little weird when an item has protection from temperatures which shows on inspect but doesn't show on the tag
should be fine

**Review by Archanial — DISMISSED** [review]

**Comment by ike709** [comment]
The sacrifices we make to not give codedivers an advantage.

**Comment by francinum** [comment]
Wow this is shit

**Pull request merged**